resolver/resolver.cc

```diff
@@ -1377,6 +1377,11 @@ class ResolveSignaturesWalk {
                         return;
                     }
                 }
+                if (ast::isa_tree<ast::UnresolvedConstantLit>(expr) || ast::isa_tree<ast::ConstantLit>(expr)) {
```

Comment:
This makes sure that we don't report the error below for constructs of the form
```
C1 = C2
```
which otherwise would get caught by this case. We previously would not have run this function in such a way that this error would have ever been reported, but because we run this now even if the `resultType` of the constant has already been resolved, we need this check.

Reply:
This is a proper error that we should report reliably instead of silently failing.

Reply:
Said otherwise, we shouldn't do cross-constant resolution as it's order dependent.